cylc/flow/data_store_mgr.py

```diff
@@ -1735,7 +1742,11 @@ def _family_ascent_point_update(self, fp_id):
                     state_counter += Counter(dict(child_node.state_totals))
             # Gather all child task states
             task_states = []
-            for tp_id in fam_node.child_tasks:
+            for tp_id in [
+                ct_id
+                for ct_id in fam_node.child_tasks
+                if not all_nodes or ct_id in all_nodes
+            ]:
```

Comment:
Can avoid intermediate list comprehension
```py
for tp_id in fam_node.child_tasks:
    if not all_nodes or tp_id in all_nodes:
```

Reply:
Which is faster?

Reply:
I mean I suppose not creating a list comprehension .. if the evaluation of the if block outside and a `continue` is faster

Reply:
I prefer the comprehension, but for performance swap the square brackets for round ones when only using the comprehension for iteration purposes, this avoides creating an intermediate list i.e:

```python
for tp_id in (
    ct_id
    ...
```

Reply:
Looks like you win Ronnie XD